### DISCUSSION TITLE
How can the source code generator generate code immediately when inputting instead of waiting until compilation?

### DISCUSSION THREAD

**Discussion by sgf** [dotnet/csharplang#7663] [Q&A] [answered]
How can the source code generator generate code immediately when inputting instead of waiting until compilation?
I searched the documentation for the source code generator and found no valuable information.

The following code has two opportunities to generate code.
If possible, I hope that in case 1, the code for Fixed10 will be generated immediately.

```C#
public struct PacketTest
{
    public Fixed10<>...  //case 1: I need generate Fixed10<T>'s code when after  input "Fixed10<> "
    public Fixed10<PData> Field1;  //case 2: I need generate Fixed10<T>'s code when after input  "public Fixed10<PData> Field1; "

}

public struct PData{
public byte Seq;
public int Len;
public byte Flags;
}

```

**Comment by siegfriedpammer** [comment]
I think this is not possible, because Roslyn source generators are by definition executed during normal compilation, not by some separate tool or the IDE.

**Comment by sgf** [comment, marked as answer]
i have soloved,
I have solved the problem. In fact, the source code generator + VS can detect code changes in near real-time. The reason why it does not work is simply because you need to compile the source code generator and then restart VS. Then wait for the project to be completely loaded, and then Switch to the corresponding code page, and then the debugger [Debugger.Launch();] will take effect.

**Reply by 333fred** [reply]
In the future, such questions are better asked on discord, https://discord.gg/csharp, in the roslyn channel, rather than here.
